lizrd/core/llm.py

```diff
@@ -325,11 +389,26 @@ def __init__(
                 layer.block_number = i_block
 
             _, current_device = self.get_current_device(i_block)
+            block = TransformerBlock(
+                dmodel,
+                layers_info,
+                gradient_checkpointing,
+                residual_fn,
+                wrap_attn_and_ff_in_fsdp=wrap_attn_and_ff_in_fsdp,
+                rank=rank,
+                param_precision=param_precision,
+                offload_params=offload_params,
+            ).to(current_device)
+            block = wrap_in_fsdp(
+                enabled=wrap_blocks_in_fsdp,
+                module=block,
+                rank=rank,
+                param_precision=param_precision,
+                offload_params=offload_params,
+            )
```

Comment:
so you first create a block and send it to target machine, and _then_ wrap it. Is this the way to go? Are there any tutorials for reference?

Reply:
Yes, this is mentioned in the documentation. In this case, when fsdp is on we always pass cpu as the device. FSDP itself puts module on the proper device, denoted by rank

Reply:
So basically the model is initialized on CPU

Reply:
Good! Thank you.